python/lsst/atmospec/centroiding.py

```diff
@@ -135,6 +136,18 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
         outputs = self.run(**inputs)
         butlerQC.put(outputs, outputRefs)
 
+    def runFallbackTask(self, exp):
+        task = self.centroidingFallbackTask
+        if isinstance(task, QuickFrameMeasurementTask):
+            result = task.run(exp)
+            return result.brightestObjCentroid  # tuple
+        elif isinstance(task, PeekExposureTask):
+            result = task.run(exp)
+            centroid = result.brightestCentroid  # Point2D
+            return (centroid[0], centroid[1])
+        else:
+            raise ValueError(f"Unsupported fallback task: {task}")
```

Comment:
Can you throw this error in `validate` method instead of during runtime? And I suppose the task won't be an instance yet.

Reply:
Yeah, that's a "this should be impossible" type raise, but you make a good point - I can add a `validate` method that checks it's one of those two, and then it actually _should be_ impossible